examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -49,20 +36,44 @@ CastingServer * CastingServer::GetInstance()
     return castingServer_;
 }
 
-void CastingServer::Init()
+CHIP_ERROR CastingServer::Init(AppParams * AppParams)
 {
     if (mInited)
     {
-        return;
+        return CHIP_NO_ERROR;
     }
 
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    // if this class's client provided a RotatingDeviceIdUniqueId, use that
+    if (AppParams != nullptr && AppParams->GetRotatingDeviceIdUniqueId().HasValue())
+    {
+        ByteSpan rotatingDeviceIdUniqueId(AppParams->GetRotatingDeviceIdUniqueId().Value());
+        chip::DeviceLayer::ConfigurationMgr().SetRotatingDeviceIdUniqueId(rotatingDeviceIdUniqueId);
+    }
+#ifdef CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID
```

Comment:
this config needs to go away. There is no situation where a compile time unique id makes sense.

Reply:
Tracking as a separate issue here: https://github.com/project-chip/connectedhomeip/issues/23305